src/ucp/wireup/address.c

```diff
@@ -618,7 +620,12 @@ ucp_address_unpack_iface_length(ucp_worker_h worker, const void *flags_ptr,
 
 uint64_t ucp_address_get_uuid(const void *address)
 {
-    return *(uint64_t*)UCS_PTR_TYPE_OFFSET(address, uint8_t);
+    const uint8_t address_header = *(const uint8_t *)address;
```

Comment:
no need to cast

Reply:
Need to check for `WORKER_UID` flag in the header